program/src/processor.rs

```diff
@@ -118,33 +118,60 @@ fn check_holder_rewards(
     Ok(())
 }
 
+// Calculate the rewards per token.
+//
+// Calculation: rewards / token_supply
+// Scaled by 1e9 to store 9 decimal places of precision.
+//
+// This calculation is valid for all possible values of rewards and token
+// supply, since the scaling to `u128` prevents multiplication from breaking
+// the `u64::MAX` ceiling, and the `token_supply == 0` check prevents
+// `checked_div` returning `None` from a zero denominator.
 fn calculate_rewards_per_token(rewards: u64, token_supply: u64) -> Result<u128, ProgramError> {
     if token_supply == 0 {
         return Ok(0);
     }
-    // Calculation: rewards / token_supply
-    //
-    // Scaled by 1e9 to store 9 decimal places of precision.
     (rewards as u128)
         .checked_mul(REWARDS_PER_TOKEN_SCALING_FACTOR)
         .and_then(|product| product.checked_div(token_supply as u128))
         .ok_or(ProgramError::ArithmeticOverflow)
 }
 
+// Calculate the eligible rewards for a token account.
+//
+// Calculation: (current - last) * balance
+// The result is descaled by a factor of 1e9 since both rewards per token
+// values are scaled by 1e9 for precision.
+//
+// This calculation is valid under the following conditions:
+// * The current accumulated rewards per token is always greater than or equal
+//   to the last accumulated rewards per token.
+// * The marginal rate multiplied by the token account balance does not exceed
+//   `u128::MAX`.
+//
+// The first condition is guaranteed by the program logic.
+// The second condition is guaranteed under the following constraints:
+// * The current supply of SOL is 500,000,000 * 1e9.
+// * The mint supply is less than or equal to 1e9.
```

Comment:
I think the idea for PAL will be to have 1 billion tokens, but with 9 decimals (just like SOL), so really 10^18 tokens